src/main/java/com/ibm/darpc/DaRPCMemPool.java

```diff
@@ -0,0 +1,435 @@
+package com.ibm.darpc;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.RandomAccessFile;
+import java.nio.ByteBuffer;
+import java.nio.MappedByteBuffer;
+import java.nio.channels.FileChannel;
+import java.nio.channels.FileChannel.MapMode;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.LinkedList;
+
+import com.ibm.disni.rdma.RdmaEndpoint;
+import com.ibm.disni.rdma.verbs.IbvMr;
+import com.ibm.disni.rdma.verbs.IbvPd;
+import com.ibm.disni.util.MemoryUtils;
+
+public class DaRPCMemPool {
+	final int defaultAllocationSize = 16 * 1024 * 1024; // 16MB
+	final int defaultMinAllocationSize = 4 * 1024; // 4KB
+	final int defaultAlignmentSize = 4 * 1024; // 4KB
+	final int defaultHugePageLimit = 0; // no huge pages by default
+
+	private HashMap<IbvPd, PdMemPool> pdMemPool; // One buddy allocator per protection domain
+	private LinkedList<IbvMr> mrs;
```

Comment:
I prefer ArrayList over LinkedList and you can use the List interface here and choose your implementation in the constructor